Documentation/MyFirstContribution

```diff
@@ -0,0 +1,584 @@
+the schema shown in the Naming Tests section of `t/README`.
+
+=== Writing Your Test
+
+Since this a toy command, let's go ahead and name the test with t9999. However,
+as many of the family/subcmd combinations are full, best practice seems to be
+to find a command close enough to the one you've added and share its naming
+space.
+
+Create your test script and mark it executable:
+
+----
+touch t/t9999-psuh-codelab.sh
+chmod +x t/t9999-psuh-codelab.sh
+----
+
+Begin with the header as so (see
+"Writing Tests" and "Source 'test-lib.sh'" in `t/README`):
+
+----
+#!/bin/sh
+
+test_description='git-psuh test
+
+This test runs git-psuh and makes sure it does not crash.'
+
+. ./test-lib.sh
+----
+
+Tests are framed inside of a `test_expect_success` in order to output TAP
+formatted results. Begin your first test and set up the repo to test in:
+
+----
+test_expect_success 'runs correctly with no args' '
+	rm -rf workbench upstream &&
+	test_create_repo upstream &&
+----
+
+`test_create_repo` comes from `test-lib.sh`. Next, we'll modify the above to
+move into the new repo and run our new command:
+
+----
+test_expect_success 'runs correctly with no args' '
+	rm -rf workbench upstream &&
+	test_create_repo upstream &&
+	(
+		cd upstream &&
+		git psuh
+	)
+'
+----
+
+Indicate that you've run everything you wanted by adding the following at the
+bottom of your script:
+
+----
+test_done
+----
+
+=== Running Locally
+
+Let's try and run locally:
+
+----
+make -j16
+cd t/ && prove t9999-psuh-codelab.sh
+----
+
+You can run the full test suite and ensure git-psuh didn't break anything:
+
+----
+cd t/
+prove -j16 --shuffle t[0-9]*.sh
+----
+
+(You can also do this with `make test` but `prove` can run concurrently.
+Shuffle randomizes the order the tests are run in, which makes them resilient
+against unwanted inter-test dependencies. `prove` also makes the output nicer.
+
+Go ahead and commit this change, as well.
+
+== Getting Ready to Share
+
+Before you send your patch off to be reviewed by the wide world, it's a good
+idea to run the Travis CI build and test suites against your new changes. You
+can do this manually or by using GitGitGadget, but either way, you're going to
+need to fork. First thing - make sure you have a GitHub account.
+
+=== Forking git/git on GitHub
+
+Head to the https://github.com/git/git[GitHub mirror] and look for the Fork
+button. Place your fork wherever you deem appropriate and create it.
+
+=== Uploading To Your Own Fork
+
+To upload your branch to your own fork, you'll need to add the new fork as a
+remote. You can use `git remote -v` to show the remotes you have added already.
+From your new fork's page on GitHub, you can press "Clone or download" to get
+the URL; then you need to run the following to add, replacing your own URL and
+remote name for the examples provided:
+
+----
+git remote add remotename git@github.com:remotename/git.git
+----
```

Comment:
This is the SSH URL, whereas my experience is that most users prefer the HTTPS URL (even for pushing)... In the least, I think we want to say his here.